### PR TITLE
feat: close client when no mapping found

### PR DESCRIPTION
**Pull request by kptdobe**
Only main enhancement here is to not execute the end of the join code when no user mapping is found.

### CODE REVIEW

**Comment by kptdobe** [comment]
Wrong approach, since the timeout method is there to handle connection loss. To fix the infinite loop, we need the "server" to respond a proper error response.

**Review by vtsaplin — APPROVED** [review]

File: src/ChatClient.ts [hunk 1]
```diff
@@ -144,23 +144,32 @@ export class ChatClient {
...
-    }
+    try {
+      const {email, channelId, teamId, channelName} = await this.sendCommand<any, any>('join', {
+        version: getAppVersion(),
+      });
+      if (!email || !channelId || !teamId || !channelName) {
+        console.error(`Client ready without email or channel. Disconnecting...`);
+        this.client.close();
+        this.fireError(new Error('No mapping found.'));
+        return;
```
Comment (vtsaplin) [inline src/ChatClient.ts:155]: Could it be that the issue with repetitive reconnection happens only due to this forgotten return?

File: src/ChatClient.ts [hunk 2]
```diff
@@ -144,23 +144,32 @@ export class ChatClient {
...
 
-    console.log(`Client ready: ${email}, ${channelId}`);
-    this.fireStatusChange(ConnectionStatus.CONNECTED);
+      console.log(`Client ready: ${email}, ${channelId}`);
+      this.fireStatusChange(ConnectionStatus.CONNECTED);
 
-    await this.requestHistory();
+      await this.requestHistory();
+    } catch (e: any) {
+      console.error(`Impossible to join. Disconnecting...`);
```
Comment (vtsaplin) [inline src/ChatClient.ts:168]: When we catch the error here, we just close the socket once again, so I am trying to understand why the fix actually works?

**Review by vtsaplin — COMMENTED** [review]
You probably fixed the problem just by adding the missing return statement. The second catch perhaps is not even necessary. wdyt?

File: src/ChatClient.ts (hunk 1 above)
Comment (kptdobe) [reply src/ChatClient.ts:155]: Actually, we never go there! The server responds an error, thus the catch is what closes the connection. But if the server would respond a 200 with an error object, we would end up here (this was the error handling I have now removed). Adding the `return` is nicer but currently totally useless.

**Pull request merged**